test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+      });
+
+      it('with options should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get({ url: TEST_SERVER_URL });
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+    });
+
+    describe('Submitting a POST request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(201, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.post(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe('');
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = { foo: 'bar' };
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody,
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(JSON.stringify(requestBody));
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = JSON.stringify({ foo: 'bar' });
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(requestBody);
+      });
+    });
+
+    describe('Submitting a PUT request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(204, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: { foo: 'bar' },
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.put({
+          url: TEST_SERVER_URL,
+          body: JSON.stringify({ foo: 'bar' })
+        });
+
+        expect(statusCode).toBe(204);
+        expect(body).toBeFalsy();
+      });
+    });
+
+    describe.skip('Submitting a PATCH request', () => {
```

Comment:
should this `.skip` be removed?

Reply:
this is related to what i brought up in the sync up: Stitch backend doesn't support `PATCH` or `HEAD` http requests at this point.  would defer to you as to how to handle this in the SDK:

* leave as is (with methods on the http service that are unsupported and skipped tests)
* remove the unsupported methods from the http service (and their skipped tests)
* implement the methods on Stitch's backend (and fill out the tests)

Reply:
Got it. Based on our sync meeting convo it sounds like we ultimately decided to implement PATCH but *not* HEAD. So let's:
- Proceed with what we have here for `patch()` (with the `.skip()` left in)
- File a ticket to add support for PATCH on the server, and include a note in the description about enabling tests for it in the JS sdk (removing `.skip()`) 
- Remove the tests/code for `head()` entirely

Reply:
We're do I just PATCH and HEAD but not OPTIONS. Ticket filed yesterday and at top of queue. My fault for having them removed initially.

Reply:
Just to tie things up here, I was able to hit Martin's latest changes to support `PATCH` and `HEAD` requests, so these are now good to go